Classes/Controller/AuthorizationController.php

```diff
@@ -0,0 +1,109 @@
+<?php
+declare(strict_types=1);
+
+namespace Flownative\Canto\Controller;
+
+/*
+ * This file is part of the Flownative.Canto package.
+ *
+ * (c) Karsten Dambekalns, Flownative GmbH - www.flownative.com
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Flownative\Canto\AssetSource\CantoAssetSource;
+use Flownative\Canto\Domain\Model\AccountAuthorization;
+use Flownative\Canto\Domain\Repository\AccountAuthorizationRepository;
+use Flownative\Canto\Service\CantoOAuthClient;
+use Flownative\OAuth2\Client\OAuthClientException;
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Log\Utility\LogEnvironment;
+use Neos\Flow\Mvc\Controller\ActionController;
+use Neos\Flow\Mvc\Exception\StopActionException;
+use Neos\Flow\Mvc\Exception\UnsupportedRequestTypeException;
+use Neos\Flow\Persistence\Exception\IllegalObjectTypeException;
+use Neos\Flow\Persistence\PersistenceManagerInterface;
+use Neos\Flow\Security\Account;
+use Neos\Flow\Security\Context;
+use Neos\Flow\Security\Exception as SecurityException;
+
+class AuthorizationController extends ActionController
+{
+    /**
+     * @Flow\Inject
+     * @var Context
+     */
+    protected $securityContext;
+
+    /**
+     * @Flow\Inject
+     * @var PersistenceManagerInterface
+     */
+    protected $persistenceManager;
+
+    /**
+     * @Flow\Inject
+     * @var AccountAuthorizationRepository
+     */
+    protected $accountAuthorizationRepository;
+
+    /**
+     * Finish OAuth2 authorization
+     *
+     * @throws OAuthClientException
+     * @throws StopActionException
+     * @throws UnsupportedRequestTypeException
+     */
+    public function finishAction(string $state, string $code, string $scope = ''): void
```

Comment:
A custom "finish authorization" action so we can redirect via HTML and not a `Location` header…